### PR TITLE
Added New Component: Floating Label

### PR DESCRIPTION
**Pull request by HariAcidReign**
**What kind of change does this PR introduce?**
Added a New Component (no external libraries used) #94 

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Not yet. Once the component is approved, I will add the documentation

**Summary**
Satisfies the feature proposal requested in https://github.com/react-native-elements/react-native-elements-universe/issues/8
- Initially, I wanted to use some third-party library to create this component. However, there was a method to implement this without any such libraries, and hence I have made the PR here and not in `React Native Elements Universe`

**Does this PR introduce a breaking change?**
No

**Other information**

https://user-images.githubusercontent.com/58134096/118386612-51b15f00-b636-11eb-9636-2713d6453049.mp4

Note: The floating label on user input goes down (instead of staying up) as I have not defined the `onChangeText` for the component. So, `props.value` will not change on text input and hence the component assumes that there is no input present, and brings the label down. However, in the user's implementation, the value will change, and hence the component will detect that `isFieldActive && !props.value` is `false` and `handleBlur` will not run that segment of code.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ import BottomSheet, { BottomSheetProps } from './bottomSheet/BottomSheet';
 import LinearProgress, { LinearProgressProps } from './linearProgress';
 import Switch, { SwitchProps } from './switch/switch';
 import Tab, { TabItemProps, TabProps } from './tab/tab';
+import FloatingInput, { FloatingInputProps } from './input/floatingInput';
```
Comment (arpitBhalla) [inline src/index.ts:46]: Wouldn't it be better if add `variant='floating'` or `variant='outlined'` Prop in input? What do you think?

Comment (HariAcidReign) [reply src/index.ts:46]: I wanted first to get the core styling of the component done and then build these new props in future iterations. that being said, that prop is indeed necessary so I'll look into it for sure.

**Review by khushal87 — COMMENTED** [review]
Please adapt the PR according to the latest changes in the repo and resolve conflicts.

**Review by khushal87 — COMMENTED** [review]
Please check the next branch for the file naming conventions and modify the files accordingly. You will find a common pattern in all the components and also a uniform directory structure. You need to modify your PR according to this. @HariAcidReign

File: src/index.ts (hunk 1 above)
Comment (flyingcircle) [reply src/index.ts]: ^ I agree with Arpit here. This seems like more of a variant rather than a completely separate new component.

**Pull request closed without merging**